.github/workflows/aws-oidc-deploy-test.yaml

```diff
@@ -0,0 +1,50 @@
+name: aws_oidc deploy test
+on:
+  workflow_dispatch: {}
+  pull_request:
+    branches: main
+    paths:
+      - 'cloud/**'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    env:
+      USE_LOCAL_BACKEND: true
+    steps:
+      - name: Check out branch
+        uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-east-1
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      - name: Resolve latest snapshot tag
+        run: source cloud/shared/bin/resolve-latest-snapshot-tag
+
+      - name: Copy the config from staging
+        run: |
+          #  Grab the config from the staging deploy repo to confirm
+          #  the changes work with actual data
+          curl https://raw.githubusercontent.com/civiform/staging-aws-deploy/main/civiform_config.sh > civiform_config.sh
+
+      - name: Source config, set CIVIFORM_MODE to be "test", Run setup with latest tag
+        run: |
+          source civiform_config.sh
+          export CIVIFORM_MODE="test"
+          export APP_PREFIX="$(git rev-parse --short HEAD)"
+          echo "${LATEST_SNAPSHOT_TAG}"
+          cloud/shared/bin/setup  "--tag=${LATEST_SNAPSHOT_TAG}"
```

Comment:
Running setup sources `civiform_config.sh`. So it will override CIVIFORM_MODE that you set previously?

Reply:
it runs https://github.com/civiform/civiform/blob/main/cloud/shared/bin/setup that does not source civiform_config.sh that's why it's sourced here explicitly.

Reply:
It runs lib.sh which in turn sources the config: https://github.com/civiform/civiform/blob/main/cloud/shared/bin/lib.sh#L16 
Though it sources config from the template folder and there is no config there AFAIU as you created it in the root directory. So that source will actually fail? Can we test this action somehow (maybe it already ran) or we need to submit and see?

Reply:
Oh, sorry, I misread the code.